internal/discover/hooks_test.go

```diff
@@ -0,0 +1,490 @@
+			expected: &cdiHookCreator{
+				nvidiaCDIHookPath: defaultNvidiaCDIHookPath,
+				fixedArgs:         []string{"nvidia-cdi-hook"},
+				disabledHooks: map[HookName]bool{
+					UpdateLDCacheHook:  true,
+					CreateSymlinksHook: true,
+					ChmodHook:          false, // Explicitly enabled
+				},
+			},
+		},
+		{
+			name: "WithDisabledHooks can be called multiple times",
+			opts: []Option{
+				WithDisabledHooks(UpdateLDCacheHook),
+				WithDisabledHooks(CreateSymlinksHook),
+				WithDisabledHooks(EnableCudaCompatHook),
+			},
+			expected: &cdiHookCreator{
+				nvidiaCDIHookPath: defaultNvidiaCDIHookPath,
+				fixedArgs:         []string{"nvidia-cdi-hook"},
+				disabledHooks: map[HookName]bool{
+					UpdateLDCacheHook:    true,
+					CreateSymlinksHook:   true,
+					EnableCudaCompatHook: true,
+					ChmodHook:            true, // Default disabled
+				},
+			},
+		},
+		{
+			name: "WithEnabledHooks overrides defaultDisabledHooks",
+			opts: []Option{
+				WithEnabledHooks(ChmodHook),
+			},
+			expected: &cdiHookCreator{
+				nvidiaCDIHookPath: defaultNvidiaCDIHookPath,
+				fixedArgs:         []string{"nvidia-cdi-hook"},
+				disabledHooks: map[HookName]bool{
+					ChmodHook: false, // ChmodHook is enabled
+				},
+			},
+		},
+		{
+			name: "nvidia-ctk binary path affects fixed args",
+			opts: []Option{
+				WithNVIDIACDIHookPath("/usr/bin/nvidia-ctk"),
+			},
+			expected: &cdiHookCreator{
+				nvidiaCDIHookPath: "/usr/bin/nvidia-ctk",
+				fixedArgs:         []string{"nvidia-ctk", "hook"},
+				disabledHooks: map[HookName]bool{
+					ChmodHook: true,
+				},
+			},
+		},
+		{
+			name: "nvidia-ctk in custom NVIDIA toolkit path",
+			opts: []Option{
+				WithNVIDIACDIHookPath("/usr/local/nvidia/toolkit/nvidia-ctk"),
+			},
+			expected: &cdiHookCreator{
+				nvidiaCDIHookPath: "/usr/local/nvidia/toolkit/nvidia-ctk",
+				fixedArgs:         []string{"nvidia-ctk", "hook"},
+				disabledHooks: map[HookName]bool{
+					ChmodHook: true,
+				},
+			},
+		},
+		{
+			name: "nvidia-cdi-hook in /usr/local/bin",
+			opts: []Option{
+				WithNVIDIACDIHookPath("/usr/local/bin/nvidia-cdi-hook"),
+			},
+			expected: &cdiHookCreator{
+				nvidiaCDIHookPath: "/usr/local/bin/nvidia-cdi-hook",
+				fixedArgs:         []string{"nvidia-cdi-hook"},
+				disabledHooks: map[HookName]bool{
+					ChmodHook: true,
+				},
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			hookCreator := NewHookCreator(tc.opts...)
+
+			require.EqualValues(t, tc.expected, hookCreator)
+		})
+	}
+}
+func TestCDIHookCreator_Create(t *testing.T) {
+	testCases := []struct {
+		name         string
+		hookCreator  HookCreator
+		hookName     HookName
+		args         []string
+		expectedHook *Hook
+	}{
+		{
+			name:        "CreateSymlinksHook with args",
+			hookCreator: NewHookCreator(WithNVIDIACDIHookPath(defaultNvidiaCDIHookPath)),
+			hookName:    CreateSymlinksHook,
+			args:        []string{"/source:/target", "/source2:/target2"},
```

Comment:
nit: The format for the link is `{{ .Source }}::{{ .Target }}`: https://github.com/NVIDIA/nvidia-container-toolkit/blob/9d2ba796a0359367efb5acbf928d4da7dc969eaf/internal/discover/symlinks.go#L35

```suggestion
			args:        []string{"/source::/target", "/source2::/target2"},
```